src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -16,36 +16,31 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug
     /// <see cref="StartupProjectRegistrar"/> is responsible for adding or removing a project from the Startup list
     /// depending on whether the active configuration of the a project is debuggable or not.
     /// </summary>
-    internal class StartupProjectRegistrar : OnceInitializedOnceDisposedAsync
+    internal class StartupProjectRegistrar : OnceInitializedOnceDisposed
     {
         private readonly IVsStartupProjectsListService _startupProjectsListService;
-        private readonly IUnconfiguredProjectVsServices _projectVsServices;
         private readonly IProjectThreadingService _threadingService;
         private readonly IActiveConfiguredProjectSubscriptionService _activeConfiguredProjectSubscriptionService;
         private readonly ActiveConfiguredProject<DebuggerLaunchProviders> _launchProviders;
 
-        private Guid _guid;
+        private Guid _guid = Guid.Empty;
         private IDisposable _evaluationSubscriptionLink;
         private bool _isDebuggable;
 
         internal DataFlowExtensionMethodCaller WrapperMethodCaller { get; set; }
 
         [ImportingConstructor]
         public StartupProjectRegistrar(
-            IUnconfiguredProjectVsServices projectVsServices,
             SVsServiceProvider serviceProvider,
             IProjectThreadingService threadingService,
             IActiveConfiguredProjectSubscriptionService activeConfiguredProjectSubscriptionService,
             ActiveConfiguredProject<DebuggerLaunchProviders> launchProviders)
-            : base(projectVsServices.ThreadingService.JoinableTaskContext)
         {
-            Requires.NotNull(projectVsServices, nameof(projectVsServices));
             Requires.NotNull(serviceProvider, nameof(serviceProvider));
             Requires.NotNull(threadingService, nameof(threadingService));
             Requires.NotNull(activeConfiguredProjectSubscriptionService, nameof(activeConfiguredProjectSubscriptionService));
             Requires.NotNull(launchProviders, nameof(launchProviders));
 
-            _projectVsServices = projectVsServices;
             _startupProjectsListService = serviceProvider.GetService<IVsStartupProjectsListService, SVsStartupProjectsListService>();
```

Comment:
I never saw this on the old code review -  you can't do this. This has to be on the UI thread. Do this in AddOrRemoveProjectFromStartupProjectList